### PR TITLE
Single results

### PR DESCRIPTION
**Pull request by janno42**
This fixes several bugs with Single Results.

### CODE REVIEW

File: evap/staff/views.py [hunk 1]
```diff
@@ -634,6 +634,9 @@ def course_delete(request):
 
     if not course.can_staff_delete:
         raise SuspiciousOperation("Deleting course not allowed")
+    if course.is_single_result:
+        for counter in RatingAnswerCounter.objects.filter(contribution__course=course):
+            counter.delete()
```
Comment (karyon) [inline evap/staff/views.py]: `RatingAnswerCounter.objects.filter(contribution__course=course).delete()`?

File: evap/results/templates/results_semester_detail.html [hunk 2]
```diff
@@ -87,8 +87,13 @@
                     </thead>
                     <tbody>
                     {% for single_result, result in single_results %}
-                        <tr>
-                            <td>{{ single_result.name }}</td>
+                        <tr{% if single_result.state == 'reviewed' %} class="preview"{% endif %}>
```
Comment (karyon) [inline evap/results/templates/results_semester_detail.html]: space before the `{%` and not inside it please

Comment (karyon) [inline evap/results/templates/results_semester_detail.html]: also, i would prefer `state != 'published'`

File: evap/results/templates/results_semester_detail.html [hunk 3]
```diff
@@ -87,8 +87,13 @@
...
+                            <td>
+                                {% if single_result.state == 'reviewed' %}
```
Comment (karyon) [inline evap/results/templates/results_semester_detail.html]: same here

Comment (janno42) [inline evap/results/templates/results_semester_detail.html]: check

File: evap/results/templates/results_semester_detail.html (hunk 2 above)
Comment (janno42) [inline evap/results/templates/results_semester_detail.html]: check

File: evap/staff/views.py (hunk 1 above)
Comment (janno42) [inline evap/staff/views.py]: check

**Comment by janno42** [comment]
i would suggest some tests, too. would you like to write them? ;)

**Comment by karyon** [comment]
no.

**Comment by janno42** [comment]
tests. i don't like them but they test stuff. creating single results is ugly.

File: evap/evaluation/tests/test_models.py [hunk 4]
```diff
@@ -116,6 +117,39 @@ def test_responsible_contributors_ordering(self):
...
+            "answer_3": 2,
+            "answer_4": 0,
+            "answer_5": 2,
+            "semester": course.semester.pk
+        }
+        form = SingleResultForm(form_data, instance=course)
+        self.assertTrue(form.is_valid())
+        form.save(user=mommy.make(UserProfile))
+
+        self.assertTrue(Course.objects.filter(pk=course.pk).exists())
```
Comment (karyon) [inline evap/evaluation/tests/test_models.py]: can you replace all that by directly creating a singleResult with model mommy? this test doesn't need to use a form at all as far as i can see.

Comment (janno42) [inline evap/evaluation/tests/test_models.py]: k

**Pull request merged**